app/services/sources/TwitterSource.java

```diff
@@ -229,4 +229,8 @@ private void setPostURL(Status s, Post.Builder builder) {
 		}
 	}
 
+	@Override
+	public String getSourceName() {
+		return TWITTER;
```

Comment:
This functionality should be provided by the access layer as it implementation specific and this requirement undocumented in the AbstractDataAccess class. Will create issue.